lib/improver/utilities/spatial.py

```diff
@@ -0,0 +1,163 @@
+        metadata.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube from which the discrete differences have been calculated.
+        coord_name : String
+            The name of the coordinate over which the discrete difference
+            have been calculated.
+        diffs_along_axis : numpy array
+            Array containing the discrete differences.
+
+        Returns
+        -------
+        discrete_diff_cube : Iris.cube.Cube
+            Cube containing the discrete differences calculated along the
+            specified axis.
+        """
+        points = cube.coord(coord_name).points
+        mean_points = (points[1:] + points[:-1]) / 2
+
+        # Copy cube metadata and coordinates into a new cube.
+        # Create a new coordinate for the coordinate along which the discrete
+        # difference has been calculated.
+        metadata_dict = copy.deepcopy(cube.metadata._asdict())
+        discrete_diff_cube = Cube(diffs_along_axis, **metadata_dict)
+
+        for coord in cube.dim_coords:
+            dims = cube.coord_dims(coord)
+            if coord.name() in [coord_name]:
+                coord = DimCoord(
+                    mean_points, standard_name=coord.standard_name,
+                    units=coord.units)
+            discrete_diff_cube.add_dim_coord(coord.copy(), dims)
+        for coord in cube.aux_coords:
+            dims = cube.coord_dims(coord)
+            discrete_diff_cube.add_aux_coord(coord.copy(), dims)
+        for coord in cube.derived_coords:
+            dims = cube.coord_dims(coord)
+            discrete_diff_cube.add_aux_coord(coord.copy(), dims)
+
+        # Add metadata to indicate that a discrete difference has been
+        # calculated.
+        cell_method = CellMethod("discrete_difference", coords=[coord_name])
+        discrete_diff_cube.add_cell_method(cell_method)
+        discrete_diff_cube.attributes["form_of_discrete_difference"] = (
+            "forward_difference")
+        return discrete_diff_cube
+
+    def calculate_discrete_difference(self, cube, coord_axis):
+        """
+        Calculate the discrete difference along the axis specified by the
+        coordinate.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube from which the discrete differences will be calculated.
+        coord_axis : String
+            Short-hand reference for the x or y coordinate, as allowed by
+            iris.util.guess_coord_axis.
+
+        Returns
+        -------
+        diff_along_cube : Iris.cube.Cube
+            Cube after the discrete differences have been calculated along the
+            specified axis.
+        """
+        coord_name = cube.coord(axis=coord_axis).name()
+        diff_axis = cube.coord_dims(coord_name)[0]
+        diffs_along_axis = np.diff(cube.data, axis=diff_axis)
+        diff_along_cube = self.create_discrete_difference_cube(
```

Comment:
`diff_along_cube` isn't a good name. `diff_cube` would be better. `diffs_along_axis` would be better as `diff_along_axis`, it matches the others better. I'm open to other names though!